crates/wasmtime/src/compile.rs

```diff
@@ -419,6 +423,27 @@ impl<'a> CompileInputs<'a> {
         ret
     }
 
+    fn clean_symbol(name: &str) -> Cow<str> {
+        // Just to be on the safe side, avoid passing user-provided data to tools
+        // like "perf" or "objdump", and filter the name.  Let only characters usually
+        // used for function names, plus some characters that might be used in name
+        // mangling.
+        let bad_char = |c: char| !c.is_alphanumeric() && !r"<>[]_-:@$".contains(c);
+        if name.chars().any(bad_char) {
+            let mut name = name
+                .chars()
+                .map(|c| if bad_char(c) { '?' } else { c })
+                .collect::<String>()
+                .trim_matches('?')
+                .trim_end_matches('?')
+                .to_string();
+            name.truncate(Self::MAX_SYMBOL_LEN);
```

Comment:
.take(N) in the iterator chain should allow the truncation to happen inline (and stop iteration early if so, since the whole chain is lazy)

Reply:
Actually, no, I'm not sure how to do this; I want to truncate the string after `trim_matches('?')` removed repeated occurrences of `'?'` in the cleaned string, not before.